core/src/main/scala/org/apache/spark/rpc/netty/Outbox.scala

```diff
@@ -215,7 +216,7 @@ private[netty] class Outbox(nettyEnv: NettyRpcEnv, val address: RpcAddress) {
     // update messages and it's safe to just drain the queue.
     var message = messages.poll()
```

Comment:
In that case the exception is no longer a singleton - it is created inside handleNetworkFailure().
Network failure should be rare but when it happens, there may be non-trivial number of messages to be notified.